crates/examples/src/bin/dwarfdump.rs

```diff
@@ -2469,3 +2487,453 @@ fn dump_macro<R: Reader, W: Write>(
+            break;
+        }
+
+        // Update current offset for next entry - reader position relative to entry pool start
+        let reader_position = entry_reader.offset_from(&unit.entry_pool_reader()?);
+        current_offset = reader_position;
+    }
+
+    Ok(series_offsets)
+}
+
+fn dump_names_by_bucket<R: Reader, W: Write>(
+    w: &mut W,
+    unit: &gimli::DebugNamesUnit<R>,
+    dwarf: &gimli::Dwarf<R>,
+) -> Result<()> {
+    // Get all the required data from the debug_names unit
+    let hash_buckets = unit.hash_buckets()?;
+    let hash_array = unit.hash_array()?;
+    let string_offsets = unit.string_offsets()?;
+    let entry_offsets = unit.entry_offsets()?;
+    let abbrev_table = unit.abbreviation_table()?;
+
+    // Following DWARF 5 Section 6.1.1.4.5 Hash Lookup Table:
+    // "All symbols that have the same index into the bucket list follow one another in the
+    // hashes array, and the indexed entry in the bucket list refers to the first symbol."
+
+    use std::collections::{BTreeMap, HashMap};
+
+    // Collect names by bucket following DWARF 5 Section 6.1.1.4.5:
+    // "All symbols that have the same index into the bucket list follow one another in the
+    // hashes array, and the indexed entry in the bucket list refers to the first symbol."
+    let mut bucket_names: BTreeMap<usize, Vec<(usize, u32, R::Offset, R::Offset)>> =
+        BTreeMap::new();
+
+    for (bucket_idx, &bucket_start) in hash_buckets.iter().enumerate() {
+        if bucket_start > 0 {
+            let start_index = (bucket_start - 1) as usize; // Convert to 0-based index
+            let mut current_index = start_index;
+
+            // Follow the chain: collect all consecutive entries that belong to this bucket
+            while current_index < hash_array.len() {
+                let current_hash = hash_array[current_index];
+                let computed_bucket = (current_hash as usize) % hash_buckets.len();
+
+                if computed_bucket == bucket_idx {
+                    // This entry belongs to our bucket
+                    bucket_names
+                        .entry(bucket_idx)
+                        .or_insert_with(|| Vec::with_capacity(2)) // Most buckets have 1-2 entries
+                        .push((
+                            current_index,
+                            current_hash,
+                            string_offsets[current_index],
+                            entry_offsets[current_index],
+                        ));
+                    current_index += 1;
+                } else {
+                    // Hit a hash from different bucket, stop chain traversal for this bucket
+                    break;
+                }
+            }
+        }
+    }
+
+    // Group entries by resolved string content to handle multiple entries per name
+    // Following DWARF 5 spec: "If two different symbol names produce the same hash value,
+    // that hash value will occur twice in the hashes array"
```

Comment:
I don't understand why we need to do this. The hash table is grouped by name, and its entries point back to the name, so we already have this grouping. Why are we building up these maps instead of simply iterating over the buckets as we display?